app/src/main/java/com/android/voyageur/model/user/UserViewModel.kt

```diff
@@ -579,39 +602,123 @@ open class UserViewModel(
    * @param reqId the request ID of the friend request to delete
    */
   fun deleteFriendRequest(reqId: String) {
-    // Optimistically update the state flow
     _sentFriendRequests.value = _sentFriendRequests.value.filterNot { it.id == reqId }
     friendRequestRepository.deleteRequest(
         reqId = reqId,
-        onSuccess = {
-          Log.d("FRIEND_REQUEST", "Friend request $reqId successfully deleted")
-          // No need to call getSentFriendRequests() since we've updated the state
-        },
+        onSuccess = { Log.d("FRIEND_REQUEST", "Friend request $reqId successfully deleted") },
         onFailure = { exception ->
           Log.e("FRIEND_REQUEST", "Failed to delete friend request: ${exception.message}")
-          // Optionally revert the state flow if needed
         })
   }
-  /** Sets up a listener for incoming friend requests so that the UI updates in real-time. */
+  /**
+   * Sets up a listener for incoming friend requests so that the UI updates in real-time. Now also
+   * triggers a notification if a new friend request arrives.
+   */
   fun listenToFriendRequests() {
     val userId = Firebase.auth.uid.orEmpty()
     if (userId.isEmpty()) return
 
-    // Remove any existing listener to prevent duplicates
     friendRequestsListener?.remove()
 
     friendRequestsListener =
         friendRequestRepository.listenToFriendRequests(
             userId = userId,
             onSuccess = { requests ->
+              val oldRequests = _friendRequests.value
               _friendRequests.value = requests
-              // Update notification count based on the number of incoming friend requests
               _notificationCount.value = requests.size.toLong()
-              // Fetch associated user data for notifications
-              getUsersByIds(requests.map { it.from }) { users -> _notificationUsers.value = users }
+
+              getUsersByIds(requests.map { it.from }) { users ->
+                _notificationUsers.value = users
+
+                // Check if there's a new request (list got bigger)
+                if (requests.size > oldRequests.size) {
+                  val oldRequestIds = oldRequests.map { it.id }.toSet()
+                  val newRequests = requests.filterNot { oldRequestIds.contains(it.id) }
+
+                  if (newRequests.isNotEmpty()) {
+                    val newRequest = newRequests.first()
+                    val senderUser = users.find { it.id == newRequest.from }
+                    val senderName = senderUser?.name ?: context?.getString(R.string.unknown)
+
```

Comment:
you can not use the context here and just use the `stringResource` file which doesnt require a context

Reply:
I do not believe that is possible since stringResource is a Composable and as such it can be called just from a Composable, but I handled this by using an auxiliary composable called AndroidStringProvider.

Reply:
Ok, then your implementation is correct. Thank you for this quick reply!